File: CRUD/PHP/Symfony3 V1/Symfony3/src/Sistema/Bundle/Controller/UsuariosController.php
<?php

// Written By Ismael Heredia in the year 2017

namespace Sistema\Bundle\Controller;

use Symfony\Bundle\FrameworkBundle\Controller\Controller;
use Symfony\Component\HttpFoundation\Request;

class UsuariosController extends Controller
{
    public function indexAction()
    {
    	if($this->get("sistema.seguridad")->validar_session()) {
            if($this->get("sistema.seguridad")->validar_session_admin()) {
                $conexion = $this->get("sistema.conexion");
                $usuarios = $conexion->buscar_usuarios("");
        		$cantidad_usuarios = count($usuarios);
            	return $this->render("SistemaBundle:Usuarios:agregar.html.twig",array("usuarios"=>$usuarios,"cantidad_usuarios"=>$cantidad_usuarios));
            } else {
                $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios","Acceso Denegado","error");
                $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                return $this->redirect($this->generateUrl("sistema_administracion"));                
            }
    	} else {
    		return $this->redirect($this->generateUrl("sistema_login"));
    	}
    }

    public function buscarUsuariosAction(Request $request) {
        if($this->get("sistema.seguridad")->validar_session()) {
            if($this->get("sistema.seguridad")->validar_session_admin()) {
                if($request->getMethod()=="POST") {
                    $patron = $request->get("nombre_buscar");
                    $conexion = $this->get("sistema.conexion");
                    $usuarios = $conexion->buscar_usuarios("");
                    $cantidad_usuarios = count($usuarios);
                    $usuarios_encontrados = $conexion->buscar_usuarios($patron);
                    $cantidad_usuarios_encontrados = count($usuarios_encontrados);
                    return $this->render("SistemaBundle:Usuarios:agregar.html.twig",
                                            array("buscador_activo"=>true,"usuarios_encontrados"=>$usuarios_encontrados,
                                                  "cantidad_usuarios_encontrados"=>$cantidad_usuarios_encontrados,"usuarios"=>$usuarios,"cantidad_usuarios"=>$cantidad_usuarios));
                } else {
                   return $this->redirect($this->generateUrl("sistema_usuarios"));
                }
            } else {
                $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios","Acceso Denegado","error");
                $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                return $this->redirect($this->generateUrl("sistema_administracion"));                  
            }
        } else {
            return $this->redirect($this->generateUrl("sistema_login"));
        }        
    }

    public function cargarEdicionUsuarioAction($id) {
        if($this->get("sistema.seguridad")->validar_session()) {
            if($this->get("sistema.seguridad")->validar_session_admin()) {
                $conexion = $this->get("sistema.conexion");
                $usuario = $conexion->cargarUsuario($id);
                $usuarios = $conexion->buscar_usuarios("");
                $cantidad_usuarios = count($usuarios);
                return $this->render("SistemaBundle:Usuarios:editar.html.twig",array("usuarios"=>$usuarios,"cantidad_usuarios"=>$cantidad_usuarios,"usuario"=>$usuario));
            } else {
                $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios","Acceso Denegado","error");
                $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                return $this->redirect($this->generateUrl("sistema_administracion"));                    
            }
        } else {
            return $this->redirect($this->generateUrl("sistema_login"));
        }
    }

    public function cargarBorradoUsuarioAction($id) {
        if($this->get("sistema.seguridad")->validar_session()) {
            if($this->get("sistema.seguridad")->validar_session_admin()) {
                $conexion = $this->get("sistema.conexion");
                $usuario = $conexion->cargarUsuario($id);
                return $this->render("SistemaBundle:Usuarios:borrar.html.twig",array("usuario"=>$usuario));
            } else {
                $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios","Acceso Denegado","error");
                $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                return $this->redirect($this->generateUrl("sistema_administracion"));                     
            }
        } else {
            return $this->redirect($this->generateUrl("sistema_login"));
        }
    }

    public function agregarUsuarioAction(Request $request) {
        if($this->get("sistema.seguridad")->validar_session()) {
            if($this->get("sistema.seguridad")->validar_session_admin()) {
                if($request->getMethod()=="POST") {

                    $conexion = $this->get("sistema.conexion");

                    $nombre_usuario = $request->get("nombre_usuario");
                    $password = $request->get("password");
                    $tipo_usuario = $request->get("tipo");
                    $fecha_registro = $this->get("sistema.funciones")->fecha_actual();

                    $mensaje = "";
                    $tipo = "";

                    if($nombre_usuario!="" && $password!="" && is_numeric($tipo_usuario) && fecha_registro!="") {
                        $password = md5($password);
                        if($conexion->agregarUsuario($nombre_usuario,$password,$tipo,$fecha_registro)) {
                            $mensaje = "Usuario agregado";
                            $tipo = "success";
                        } else {
                            $mensaje = "Ha ocurrido un error en la base de datos";
                            $tipo = "error";
                        }
                    } else {
                        $mensaje = "Faltan datos";
                        $tipo = "warning"; 
                    }

                    $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios",$mensaje,$tipo);
                    $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);

                    return $this->redirect($this->generateUrl("sistema_usuarios"));
                } else {
                    return $this->redirect($this->generateUrl("sistema_usuarios"));
                }
            } else {
                $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios","Acceso Denegado","error");
                $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                return $this->redirect($this->generateUrl("sistema_administracion"));                  
            }
        } else {
            return $this->redirect($this->generateUrl("sistema_login"));
        }
    }

    public function editarUsuarioAction(Request $request) {
        if($this->get("sistema.seguridad")->validar_session()) {
            if($this->get("sistema.seguridad")->validar_session_admin()) {
                if($request->getMethod()=="POST") {

                    $conexion = $this->get("sistema.conexion");

                    $id_usuario = $request->get("id_usuario");
                    $tipo_usuario = $request->get("tipo");

                    $mensaje = "";
                    $tipo = "";

                    if(is_numeric($id_usuario) && is_numeric($tipo_usuario)) {
                        if($conexion->asignarUsuario($id_usuario,$tipo_usuario)) {
                            $mensaje = "Usuario editado";
                            $tipo = "success";
                        } else {
                            $mensaje = "Ha ocurrido un error en la base de datos";
                            $tipo = "error";
                        }
                    } else {
                        $mensaje = "Faltan datos";
                        $tipo = "warning"; 
                    }

                    $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios",$mensaje,$tipo);
                    $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);

                    return $this->redirect($this->generateUrl("sistema_usuarios"));
                } else {
                    return $this->redirect($this->generateUrl("sistema_usuarios"));
                }
            } else {
                $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios","Acceso Denegado","error");
                $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                return $this->redirect($this->generateUrl("sistema_administracion"));                  
            }
        } else {
            return $this->redirect($this->generateUrl("sistema_login"));
        }
    }

    public function borrarUsuarioAction(Request $request) {
        if($this->get("sistema.seguridad")->validar_session()) {
            if($this->get("sistema.seguridad")->validar_session_admin()) {
                if($request->getMethod()=="POST") {
                    if ($request->request->has("borrar_usuario")) {

                        $conexion = $this->get("sistema.conexion");

                        $id_usuario = $request->get("id_usuario");

                        $mensaje = "";
                        $tipo = "";

                        if(is_numeric($id_usuario)) {
                            if($conexion->borrarUsuario($id_usuario)) {
                                $mensaje = "Usuario borrado";
                                $tipo = "success";
                            } else {
                                $mensaje = "Ha ocurrido un error en la base de datos";
                                $tipo = "error";
                            }
                        } else {
                            $mensaje = "ID Invalido";
                            $tipo = "warning";
                        }

                        $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios",$mensaje,$tipo);
                        $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                        return $this->redirect($this->generateUrl("sistema_usuarios"));
                    }
                    elseif ($request->request->has("volver_usuario")) {
                        return $this->redirect($this->generateUrl("sistema_usuarios"));
                    } else {
                        return $this->redirect($this->generateUrl("sistema_usuarios")); 
                    }
                } else {
                    return $this->redirect($this->generateUrl("sistema_usuarios"));
                }
            } else {
                $mensaje_listo = $this->get("sistema.funciones")->mensaje("Usuarios","Acceso Denegado","error");
                $this->get("session")->getFlashBag()->add("mensaje",$mensaje_listo);
                return $this->redirect($this->generateUrl("sistema_administracion"));                  
            }
        } else {
            return $this->redirect($this->generateUrl("sistema_login"));
        }
    }

}

File: CRUD/Java usando NetBeans/Aplicacion Web usando MVC Spring/SistemaWebSpring/src/main/java/sistema/controllers/UsuariosController.java
// Written by Ismael Heredia in the year 2016

package sistema.controllers;

import java.util.ArrayList;
import javax.servlet.http.HttpServletRequest;
import org.springframework.context.ApplicationContext;
import org.springframework.context.support.ClassPathXmlApplicationContext;
import org.springframework.stereotype.Controller;
import org.springframework.web.bind.annotation.ModelAttribute;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.servlet.ModelAndView;
import sistema.dao.UsuarioDAO;
import sistema.functions.Funciones;
import sistema.models.Buscador;
import sistema.models.Mensaje;
import sistema.models.Usuario;

@Controller
public class UsuariosController {

    ApplicationContext context
            = new ClassPathXmlApplicationContext("Spring-Module.xml");

    UsuarioDAO usuarioDAO = (UsuarioDAO) context.getBean("UsuarioDAO");

    Funciones funcion = new Funciones();

    @RequestMapping("/administracion/usuarios")
    public ModelAndView UsuariosForm(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                Buscador buscador = new Buscador();
                Usuario usuario = new Usuario();
                int cantidad_usuarios_total = usuarioDAO.listUsuarios("").size();
                ArrayList usuarios = usuarioDAO.listUsuarios("");
                ModelAndView model = new ModelAndView("FormAgregarUsuario");
                model.addObject("cantidad_usuarios_total", cantidad_usuarios_total);
                model.addObject("Buscador", buscador);
                model.addObject("BuscadorActivo", "0");
                model.addObject("Usuario", usuario);
                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping("/administracion/usuarios/buscar")
    public ModelAndView FormBuscadorUsuariosGET(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                return new ModelAndView("redirect:/administracion/usuarios");
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/buscar", method = RequestMethod.POST)
    public ModelAndView FormBuscadorUsuarios(@ModelAttribute("Buscador") Buscador datos, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                Buscador buscador = new Buscador();
                Usuario usuario = new Usuario();
                String nombre_buscar = datos.getNombre_buscar();
                int cantidad_usuarios_total = usuarioDAO.listUsuarios("").size();
                ArrayList usuarios = usuarioDAO.listUsuarios(nombre_buscar);
                int cantidad_usuarios_encontrados = usuarios.size();
                ModelAndView model = new ModelAndView("FormAgregarUsuario");
                model.addObject("cantidad_usuarios_total", cantidad_usuarios_total);
                model.addObject("cantidad_usuarios_encontrados", cantidad_usuarios_encontrados);
                model.addObject("Buscador", buscador);
                model.addObject("BuscadorActivo", "1");
                model.addObject("Usuario", usuario);
                model.addObject("usuarios", usuarios);
                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping("/administracion/usuarios/agregar")
    public ModelAndView FormAgregarUsuarioGET(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                return new ModelAndView("redirect:/administracion/usuarios");
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/agregar", method = RequestMethod.POST)
    public ModelAndView FormAgregarUsuario(@ModelAttribute("Usuario") Usuario usuario, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String nombre = usuario.getNombre();
                String password = funcion.md5_encode(usuario.getPassword());
                int tipo_usuario = usuario.getTipo();
                String fecha_registro = funcion.fecha_del_dia();
                usuario.setPassword(funcion.md5_encode(usuario.getPassword()));
                usuario.setFecha_registro(fecha_registro);

                String msg = "";
                String tipo = "";

                if (!"".equals(nombre) && !"".equals(password) && funcion.valid_number(String.valueOf(tipo_usuario)) && !"".equals(fecha_registro)) {
                    if (usuarioDAO.comprobar_existencia_usuario_crear(nombre)) {
                        msg = "El usuario " + nombre + " ya existe";
                        tipo = "warning";
                    } else {
                        if (usuarioDAO.insert(usuario)) {
                            msg = "Usuario registrado";
                            tipo = "success";
                        } else {
                            msg = "Ha ocurrido un error en la base de datos";
                            tipo = "error";
                        }
                    }
                } else {
                    msg = "Faltan datos";
                    tipo = "warning";
                }

                Mensaje mensaje = new Mensaje("Usuarios", msg, tipo, "administracion/usuarios");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);

                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }

        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/{id}/editar", method = RequestMethod.GET)
    public ModelAndView FormEditandoUsuario(@PathVariable int id, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String msg = "";
                String tipo = "";
                if (id > 0) {
                    Usuario usuario = usuarioDAO.findByUsuarioId(id);
                    Buscador buscador = new Buscador();
                    int cantidad_usuarios_total = usuarioDAO.listUsuarios("").size();
                    ModelAndView model = new ModelAndView("FormEditarUsuario");
                    model.addObject("cantidad_usuarios_total", cantidad_usuarios_total);
                    model.addObject("Buscador", buscador);
                    model.addObject("BuscadorActivo", "0");
                    model.addObject("Usuario", usuario);
                    return model;
                } else {
                    Mensaje mensaje = new Mensaje("Usuarios", "ID Invalido", "warning", "administracion/usuarios");
                    ModelAndView model = new ModelAndView("Redireccion");
                    model.addObject("mensaje", mensaje);
                    return model;
                }
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping("/administracion/usuarios/editar")
    public ModelAndView FormEditarUsuarioGET(HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                return new ModelAndView("redirect:/administracion/usuarios");
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/editar", method = RequestMethod.POST)
    public ModelAndView FormEditarUsuario(@ModelAttribute("Usuario") Usuario usuario, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                int id_usuario = usuario.getId_usuario();
                String nombre = usuario.getNombre();
                String password = usuario.getPassword();
                int tipo_usuario = usuario.getTipo();

                String msg = "";
                String tipo = "";

                if (funcion.valid_number(String.valueOf(id_usuario)) && !"".equals(nombre) && funcion.valid_number(String.valueOf(tipo_usuario))) {
                    if (usuarioDAO.update(usuario)) {
                        msg = "Usuario editado";
                        tipo = "success";
                    } else {
                        msg = "Ha ocurrido un error en la base de datos";
                        tipo = "error";
                    }
                } else {
                    msg = "Faltan datos";
                    tipo = "warning";
                }

                Mensaje mensaje = new Mensaje("Usuarios", msg, tipo, "administracion/usuarios");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);

                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }

        } else {
            return new ModelAndView("redirect:/login");
        }
    }

    @RequestMapping(value = "/administracion/usuarios/{id}/borrar", method = RequestMethod.GET)
    public ModelAndView FormBorrarUsuario(@PathVariable int id, HttpServletRequest request) {
        if (funcion.valid_cookie(request)) {
            if (funcion.validar_cookie_admin(request)) {
                String msg = "";
                String tipo = "";
                if (id > 0) {
                    if (usuarioDAO.delete(usuarioDAO.findByUsuarioId(id)) == true) {
                        msg = "Usuario borrado";
                        tipo = "success";
                    } else {
                        msg = "Ha ocurrido un error en la base de datos";
                        tipo = "error";
                    }
                } else {
                    msg = "ID Invalido";
                    tipo = "warning";
                }
                Mensaje mensaje = new Mensaje("Usuarios", msg, tipo, "administracion/usuarios");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);

                return model;
            } else {
                Mensaje mensaje = new Mensaje("Usuarios", "Acceso Denegado", "error", "administracion");
                ModelAndView model = new ModelAndView("Redireccion");
                model.addObject("mensaje", mensaje);
                return model;
            }
        } else {
            return new ModelAndView("redirect:/login");
        }
    }

}
